src/ducks/rates.js

```diff
@@ -0,0 +1,69 @@
+import { createSlice } from '@reduxjs/toolkit';
+import snxJSConnector from '../utils/snxJSConnector';
+import { bigNumberFormatter, parseBytes32String } from '../utils/formatters';
+
+export const ratesSlice = createSlice({
+	name: 'rates',
+	initialState: {
+		exchangeRates: null,
+		loadingError: null,
+		isLoading: false,
+		isLoaded: false,
+		isRefreshing: false,
+		isRefreshed: false,
+	},
+	reducers: {
+		fetchRatesRequest: state => {
+			state.loadingError = null;
+			state.isLoading = true;
+			if (state.isLoaded) {
+				state.isRefreshing = true;
+			}
+		},
+		fetchRatesFailure: (state, action) => {
+			state.loadingError = action.payload.error;
+			state.isLoading = false;
+			state.isRefreshing = false;
+		},
+		fetchRatesSuccess: (state, action) => {
+			state.exchangeRates = action.payload.exchangeRates;
+			state.isLoading = false;
+			state.isRefreshing = false;
+			state.isLoaded = true;
+		},
+	},
+});
+
+export const getRatesState = state => state.rates;
+export const getIsLoadingRates = state => getRatesState(state).isLoading;
+export const getIsRefreshingRates = state => getRatesState(state).isRefreshing;
+export const getIsLoadedRates = state => getRatesState(state).isLoaded;
+export const getRatesLoadingError = state => getRatesState(state).loadingError;
+export const getRatesExchangeRates = state => getRatesState(state).exchangeRates;
+export const getEthRate = state => {
```

Comment:
This is a computed selector, so it should be wrapped with `createSelector`

```
import get from 'lodash/get'
import { SYNTHS_MAP } from '../../constants/currency';

export const getEthRate = createSelector(getRatesExchangeRates, exchangeRates =>
	get(exchangeRates, SYNTHS_MAP.sETH, null)
);
```

Obviously, this can be implemented without `get`, but the important bit is using reselect for this.

This will make sure to return `null` in all cases (even if `sETH` is undefined for some reason)

Reply:
right, I was wondering if this implementation was OK or not actually